src/App.tsx

```diff
@@ -1,223 +1,188 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import {
+  useState,
+  useContext,
+  useEffect,
+  useRef,
+  FormEvent,
+  useMemo,
+  useCallback,
+} from 'react';
 import { AuthContext } from './components/Auth/AuthContext';
+import { Header } from './components/Header/Header';
+import { Footer } from './components/Footer/Footer';
+import { FilterStatus } from './types/FilterStatus';
+import { ErrorNotification } from './components/ErrorNotification';
+import { TodosList } from './components/TodosList/TodosList';
+import {
+  createTodo, deleteTodo, getTodos, updateTodo,
+} from './api/todos';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
+  const user = useContext(AuthContext);
+  const [title, setTitle] = useState('');
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterType, setFilterType] = useState<FilterStatus>(FilterStatus.All);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [isTodoLoaded, setIsTodoLoaded] = useState(false);
+  const [selectedTodos, setSelectedTodos] = useState<number[]>([]);
+
+  const filteredTodos = useMemo(() => {
+    return todos.filter(todo => {
+      switch (filterType) {
+        case FilterStatus.All:
+          return todo;
+
+        case FilterStatus.Active:
+          return !todo.completed;
+
+        case FilterStatus.Completed:
+          return todo.completed;
+
+        default:
+          return true;
+      }
+    });
+  }, [todos, filterType]);
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
+
+    const fetchData = async () => {
+      const todosFromServer = await getTodos(user?.id || 0);
+
+      setTodos(todosFromServer);
+    };
+
+    try {
+      fetchData();
+    } catch {
+      setErrorMessage('Unable to connect to the server');
+    }
   }, []);
 
+  useEffect(() => {
+    if (newTodoField.current) {
+      newTodoField.current.focus();
+    }
+  }, [todos]);
+
+  const handleSubmit = async (event: FormEvent) => {
+    event.preventDefault();
+
+    if (!title.trim()) {
+      setErrorMessage("Title can't be empty");
+
+      return;
+    }
+
+    setIsTodoLoaded(true);
+
+    try {
+      const newTodo = await createTodo(user?.id || 0, title);
+
+      setTodos([...todos, newTodo]);
+    } catch {
+      setErrorMessage('Unable to add a todo');
+    } finally {
+      setTitle('');
+      setIsTodoLoaded(false);
+    }
+  };
+
+  const handleRemove = async (id: number) => {
+    setSelectedTodos([id]);
+    try {
+      await deleteTodo(id);
+
+      setTodos(todos.filter(todo => todo.id !== id));
```

Comment:
```suggestion
  const handleRemove = async (id: number) => {
    setSelectedTodos([...selectedTodos, id]);
    try {
      await deleteTodo(id);

      setTodos(todos.filter(todo => todo.id !== id));
```

try this for multiply deleting ↑